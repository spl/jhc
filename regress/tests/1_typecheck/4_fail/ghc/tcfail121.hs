
{-# LANGUAGE OverlappingInstances, FlexibleInstances #-}

module ShouldFail where

class Foo a where
  op :: a -> a

instance Foo a => Foo [a] 
instance Foo [Int]

foo :: Foo a => [a] -> [a]
foo x = op x
-- Correct instance depends on instantiation of 'a' 
