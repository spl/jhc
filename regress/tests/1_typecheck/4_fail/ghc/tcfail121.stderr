
tcfail121.hs:13:9:
    Overlapping instances for Foo [a]
      arising from a use of `op'
    Matching instances:
      instance [overlap ok] Foo a => Foo [a]
        -- Defined at tcfail121.hs:9:10
      instance [overlap ok] Foo [Int] -- Defined at tcfail121.hs:10:10
    (The choice depends on the instantiation of `a'
     To pick the first instance above, use -XIncoherentInstances
     when compiling the other instance declarations)
    In the expression: op x
    In an equation for `foo': foo x = op x
